server/src/main/java/io/unitycatalog/server/persist/StagingTableRepository.java

```diff
@@ -125,4 +156,49 @@ public StagingTableInfo createStagingTable(CreateStagingTable createStagingTable
         "Error creating table: " + getTableFullName(createStagingTable),
         /* readOnly = */ false);
   }
+
+  /**
+   * Commits a staging table by marking it as finalized and associating it with a permanent table.
+   *
+   * <p>This method is called by TableRepository.createTable when a table is created using a staging
+   * table location. It validates that the caller is the owner of the staging table and that the
+   * staging table has not been previously committed.
+   *
+   * @param session the Hibernate session for database operations
+   * @param callerId the identifier of the user attempting to commit the staging table
+   * @param storageLocation the storage location URI of the staging table to commit
+   * @return StagingTableDAO the committed staging table data access object
+   * @throws BaseException with ErrorCode.NOT_FOUND if no staging table exists at the specified
+   *     location
+   * @throws BaseException with ErrorCode.PERMISSION_DENIED if the caller is not the owner of the
+   *     staging table
+   * @throws BaseException with ErrorCode.FAILED_PRECONDITION if the staging table has already been
+   *     committed
+   */
+  public StagingTableDAO commitStagingTable(
+      Session session, String callerId, String storageLocation) {
+    serverProperties.checkManagedTableEnabled();
+    StagingTableDAO stagingTableDAO = findByStagingLocation(session, storageLocation);
+    if (stagingTableDAO == null) {
+      throw new BaseException(ErrorCode.NOT_FOUND, "Staging table not found: " + storageLocation);
+    }
+    if (!Objects.equals(stagingTableDAO.getCreatedBy(), callerId)) {
+      throw new BaseException(
+          ErrorCode.PERMISSION_DENIED,
+          "User attempts to create table on a staging location without ownership: "
+              + storageLocation);
+    }
+    if (stagingTableDAO.isStageCommitted()) {
+      throw new BaseException(
+          ErrorCode.FAILED_PRECONDITION, "Staging table already committed: " + storageLocation);
+    }
+
+    // Commit the staging table
+    Date now = new Date();
+    stagingTableDAO.setStageCommitted(true);
+    stagingTableDAO.setStageCommittedAt(now);
+    stagingTableDAO.setAccessedAt(now);
+    session.merge(stagingTableDAO);
```

Comment:
What will happen if two contention competitors are committing the same `stagingTable` at the same time  ?